### DISCUSSION TITLE
Accessing Channel metadata

### DISCUSSION THREAD

**Discussion by elavy-harris** [nextgenhealthcare/connect#5253] [Q&A]
Is there a way to access the metadata for a channel from within a transformer or filter?  More specifically, if I'm in a Channel Writer's destination transformer, can I get the target channelId for the specific destination I'm processing?

**Comment by jonbartels** [comment]
>  can I get the target channelId for the specific destination I'm processing?

Yes. Just type `channelId` or `channelName`. It'll show up in blue as a keyword and it's a string.

**Reply by elavy-harris** [reply]
Isn't that MY channelId?  I'm looking for the channelId that the Destination is going to send to.

**Reply by jonbartels** [reply]
Ooops! I misread the problem.

Is the destination channel ID hardcoded in the channel or are you referencing a mapped value for it?

**Reply by elavy-harris** [reply]
Probably hard-coded in the channel _destination,_ but if there's a way to handle mapped values, it would be nice to be able to do so.

**Comment by tonygermano** [comment]
It's not easy to access destination connector properties from the destination transformer. You can do it using either the internal (non-public) controllers or via the REST API (but this will give you the last **saved** value, rather than the currently deployed value if they happen to be different.)

The Channel Writer does support using a map variable containing the channelId if you want to set it elsewhere (configurationMap, globalChannelMap in the channel deploy script, channelMap or connectorMap in the destination transformer itself, etc...)
